### ISSUE TITLE
[FIX] List of minor fixes to be made to the development branch

### ISSUE THREAD

**Issue by ravinkohli** [automl/Auto-PyTorch#301]
These are some of the minor fixes to be made to the development branch. 

- [ ] N_threads in API module. 
- [ ] Dropout Shape in ShapedResNetBackbone. [these](https://github.com/automl/Auto-PyTorch/blob/43d4639dc84e2657662586996f0002c98a331766/autoPyTorch/pipeline/components/setup/network_backbone/ShapedResNetBackbone.py#L48-L55) lines. 
- [ ] Fixes in shake drop and shake-shake utils. See [this](https://github.com/automl/Auto-PyTorch/blob/43d4639dc84e2657662586996f0002c98a331766/autoPyTorch/pipeline/components/setup/network_backbone/utils.py#L143) and [this](https://github.com/automl/Auto-PyTorch/blob/43d4639dc84e2657662586996f0002c98a331766/autoPyTorch/pipeline/components/setup/network_backbone/utils.py#L197-L201)
- [ ] Fixes in fully connected head config space. See [this](https://github.com/automl/Auto-PyTorch/blob/43d4639dc84e2657662586996f0002c98a331766/autoPyTorch/pipeline/components/setup/network_head/fully_connected.py#L62-L94) 
- [ ] Xaviar_normal in Xavier. [Here](https://github.com/automl/Auto-PyTorch/blob/43d4639dc84e2657662586996f0002c98a331766/autoPyTorch/pipeline/components/setup/network_initializer/XavierInit.py#L27)
- [ ] Add test loader in base_data_loader and add random state. [Here](https://github.com/automl/Auto-PyTorch/blob/43d4639dc84e2657662586996f0002c98a331766/autoPyTorch/pipeline/components/training/data_loader/base_data_loader.py#L134-L152)
- [ ] Predict_proba in pipeline tabular classification. [Here](https://github.com/automl/Auto-PyTorch/blob/43d4639dc84e2657662586996f0002c98a331766/autoPyTorch/pipeline/tabular_classification.py#L93-L105)
